### PR TITLE
Apply default llvm format

### PR DESCRIPTION
**Pull request by bmourat**
This PR changes .clang-format to be completely similar to llvm style and applies this style to source code base.
There are a lot of discrepancies with our current formatting style, so I wonder if we really need this?

### CODE REVIEW

File: AppCenter/AppCenter/Internals/Device/MSDeviceTracker.m [hunk 1]
```diff
@@ -292,35 +316,47 @@ - (NSString *)osBuild {
...
    * In addition to that;
-   *     1. preferred language returns "-" instead of "_" as a delimiter of language code and country code, the method
-   * will concatenate language code and country code with "_" and return it.
-   *     2. some languages can be set without country code so region code can be returned in this case.
-   *     3. some langugaes have script code which differentiate languages. E.g. zh-Hans and zh-Hant. This is a possible
-   * scenario in Apple platforms that a locale can be zh_CN for Traditional Chinese. The method will return zh-Hant_CN
-   * in this case to make sure system language is Traditional Chinese even though region is set to China.
+   *     1. preferred language returns "-" instead of "_" as a delimiter of
+   * language code and country code, the method will concatenate language code
+   * and country code with "_" and return it. 2. some languages can be set
```
Comment (MatkovIvan) [inline AppCenter/AppCenter/Internals/Device/MSDeviceTracker.m]: Broken format

File: AppCenter/AppCenter/Internals/Ingestion/MSIngestionCall.m [hunk 2]
```diff
@@ -68,28 +73,34 @@ - (void)resetRetry {
...
   BOOL internetIsDown = [MSIngestionUtil isNoInternetConnectionError:error];
-  BOOL couldNotEstablishSecureConnection = [MSIngestionUtil isSSLConnectionError:error];
+  BOOL couldNotEstablishSecureConnection =
+      [MSIngestionUtil isSSLConnectionError:error];
 
   if (internetIsDown || couldNotEstablishSecureConnection) {
 
-    // Reset the retry count, will retry once the (secure) connection is established again.
+    // Reset the retry count, will retry once the (secure) connection is
+    // established again.
```
Comment (MatkovIvan) [inline AppCenter/AppCenter/Internals/Ingestion/MSIngestionCall.m]: Should we use `/* */` in the places like this?

File: AppCenter/AppCenter/Internals/Ingestion/Util/MSIngestionUtil.h [hunk 3]
```diff
@@ -35,8 +37,9 @@ static NSString *const kMSHidingStringForAppSecret = @"*";
 + (BOOL)isNoInternetConnectionError:(NSError *)error;
 
 /**
- * Indicate if error is because a secure connection could not be established, e.g. when using a public network that
- * is open but requires accepting terms and conditions, and the user hasn't done that, yet.
+ * Indicate if error is because a secure connection could not be established,
+ e.g. when using a public network that * is open but requires accepting terms
```
Comment (MatkovIvan) [inline AppCenter/AppCenter/Internals/Ingestion/Util/MSIngestionUtil.h]: Missing `*` at line start

File: AppCenter/AppCenter/Internals/Model/MSDeviceInternal.h [hunk 4]
```diff
@@ -81,8 +83,8 @@
 @property(nonatomic, copy) NSString *appBuild;
 
 /*
- * The bundle identifier, package identifier, or namespace, depending on what the individual plattforms use,  .e.g
- * com.microsoft.example.  [optional]
+ * The bundle identifier, package identifier, or namespace, depending on what
+ * the individual plattforms use,  .e.g com.microsoft.example.  [optional]
```
Comment (MatkovIvan) [inline AppCenter/AppCenter/Internals/Model/MSDeviceInternal.h]: Double space

File: AppCenter/AppCenter/Internals/Model/Util/MSUserDefaults.m [hunk 5]
```diff
@@ -72,7 +77,11 @@ - (NSDictionary *)updateDictionary:(NSDictionary *)dict forKey:(NSString *)key {
 }
 
 - (BOOL)updateObject:(id)o forKey:(NSString *)key expiration:(float)expiration {
-  NSDictionary *update = [self updateDictionary:@{ @"v" : o } forKey:key expiration:expiration];
+  NSDictionary *update = [self updateDictionary:@{
+    @"v" : o
+  }
+                                         forKey:key
+                                     expiration:expiration];
```
Comment (MatkovIvan) [inline AppCenter/AppCenter/Internals/Model/Util/MSUserDefaults.m]: Formatting looks wrong here, should we really do it like this?

File: AppCenter/AppCenter/Internals/Util/MSUtility.m [hunk 6]
```diff
@@ -15,21 +15,26 @@
...
 static ms_info_t appcenter_library_info
-    __attribute__((section("__TEXT,__ms_ios,regular,no_dead_strip"))) = {.info_version = 1,
-                                                                         .ms_name = APP_CENTER_C_NAME,
-                                                                         .ms_version = APP_CENTER_C_VERSION,
-                                                                         .ms_build = APP_CENTER_C_BUILD};
+    __attribute__((section("__TEXT,__ms_ios,regular,no_dead_strip"))) = {
+        .info_version = 1,
+        .ms_name = APP_CENTER_C_NAME,
+        .ms_version = APP_CENTER_C_VERSION,
+        .ms_build = APP_CENTER_C_BUILD};
```
Comment (MatkovIvan) [inline AppCenter/AppCenter/Internals/Util/MSUtility.m]: Why there is no new line before `}`?

File: AppCenter/AppCenter/MSAppCenterErrors.h [hunk 7]
```diff
@@ -17,7 +17,8 @@ extern NSString const *kMSACLogInvalidContainerErrorDesc;
 #pragma mark - Connection
 
 // Error codes
-NS_ENUM(NSInteger){kMSACConnectionSuspendedErrorCode = 100, kMSACConnectionHttpErrorCode = 101};
+NS_ENUM(NSInteger){kMSACConnectionSuspendedErrorCode = 100,
+                   kMSACConnectionHttpErrorCode = 101};
```
Comment (MatkovIvan) [inline AppCenter/AppCenter/MSAppCenterErrors.h]: Let's do it like
```
{
  kMSACConnectionSuspendedErrorCode = 100,
  kMSACConnectionHttpErrorCode = 101
};
```

File: AppCenter/AppCenter/MSEnable.h [hunk 8]
```diff
@@ -10,8 +10,10 @@
 /**
  *  Enable/disable this instance and delete data on disabled state.
  *
- *  @param isEnabled  A boolean value set to YES to enable the instance or NO to disable it.
- *  @param deleteData A boolean value set to YES to delete data or NO to keep it.
+ *  @param isEnabled  A boolean value set to YES to enable the instance or NO to
+ * disable it.
+ *  @param deleteData A boolean value set to YES to delete data or NO to keep
```
Comment (MatkovIvan) [inline AppCenter/AppCenter/MSEnable.h]: Double space at the begin of line

File: AppCenter/AppCenterTests/MSAppCenterIngestionTests.m [hunk 9]
```diff
@@ -147,46 +164,57 @@ - (void)testNetworkDown {
...
 
   // When
   [self.sut sendAsync:container
               appSecret:kMSTestAppSecret
-      completionHandler:^(__attribute__((unused)) NSString *batchId, __attribute__((unused)) NSUInteger statusCode,
-                          __attribute__((unused)) NSData *data, __attribute__((unused)) NSError *error) {
+      completionHandler:^(__attribute__((unused)) NSString *batchId,
+                          __attribute__((unused)) NSUInteger statusCode,
+                          __attribute__((unused)) NSData *data,
+                          __attribute__((unused)) NSError *error) {
```
Comment (MatkovIvan) [inline AppCenter/AppCenterTests/MSAppCenterIngestionTests.m]: This is good change to change it to `__unused`

File: AppCenter/AppCenter/Internals/Ingestion/MSIngestionCall.m (hunk 2 above)
Comment (TroubleMakerBen) [reply AppCenter/AppCenter/Internals/Ingestion/MSIngestionCall.m]: Ideally, yes. Will this work with default LLVM format?

File: AppCenter/AppCenter/Internals/Model/Util/MSUserDefaults.m (hunk 5 above)
Comment (TroubleMakerBen) [reply AppCenter/AppCenter/Internals/Model/Util/MSUserDefaults.m]: It looks weird, I agree, but code formatting seems do do it like this.

File: .clang-format [hunk 10]
```diff
@@ -1,3 +1 @@
 BasedOnStyle: LLVM
```
Comment (TroubleMakerBen) [inline .clang-format]: We can get rid of the entire file. The XcodeFormatter doesn't need a custom one.

File: AppCenter/AppCenter/Internals/Ingestion/MSOneCollectorIngestion.m [hunk 11]
```diff
@@ -42,25 +48,34 @@ - (void)sendAsync:(NSObject *)data
...
   /*
-   * FIXME: All logs are already validated at the time the logs are enqueued to Channel. It is not necessary but it can
-   * still protect against invalid logs being sent to server that are messed up somehow in Storage. If we see
-   * performance issues due to this validation, we will remove `[container isValid]` call below.
+   * FIXME: All logs are already validated at the time the logs are enqueued to
+   * Channel. It is not necessary but it can still protect against invalid logs
+   * being sent to server that are messed up somehow in Storage. If we see
+   * performance issues due to this validation, we will remove `[container
+   * isValid]` call below.
    */
```
Comment (TroubleMakerBen) [inline AppCenter/AppCenter/Internals/Ingestion/MSOneCollectorIngestion.m]: add empty line?

File: TestApps/SasquatchTV/SasquatchTVObjC/main.m [hunk 12]
```diff
@@ -1,8 +1,9 @@
-#import <UIKit/UIKit.h>
 #import "AppDelegate.h"
```
Comment (TroubleMakerBen) [inline TestApps/SasquatchTV/SasquatchTVObjC/main.m]: We're finally following convention here.

File: AppCenter/AppCenter/Internals/Device/MSDeviceTracker.m (hunk 1 above)
Comment (MatkovIvan) [reply AppCenter/AppCenter/Internals/Device/MSDeviceTracker.m]: Not fixed. Here is broken list. Please add new lines before numbers.

File: AppCenter/AppCenter/MSAppCenterErrors.h (hunk 7 above)
Comment (TroubleMakerBen) [reply AppCenter/AppCenter/MSAppCenterErrors.h]: Agreed!

**Pull request merged**